client/packages/system/src/Encounter/DetailView/DetailView.tsx

```diff
@@ -25,22 +25,25 @@ export const DetailView: FC = () => {
     encounter?.type ?? ''
   );
 
-  const { JsonForm, saveData, isDirty, validationError, revert } = useJsonForms(
-    encounter?.document?.name,
-    {
-      handleSave,
-    }
-  );
+  const {
+    JsonForm,
+    data,
+    setData,
+    saveData,
+    isDirty,
+    validationError,
+    revert,
+  } = useJsonForms(encounter?.document?.name, {
+    handleSave,
+  });
 
   const updateEncounter = useDebounceCallback(
     (patch: Partial<EncounterFragment>) =>
-      handleSave?.(
-        { ...encounter?.document.data, ...patch },
-        encounter?.document?.documentRegistry?.formSchemaId ?? '',
-        encounter?.document.id
-      ),
-    [encounter],
-    1000
+      setData({
+        ...data,
+        ...patch,
+      }),
+    [encounter, data, setData]
```

Comment:
encounter is unused - and won't `data` be updated when `encounter` is anyway?

Reply:
thanks, fixed it